policy-controller/k8s/status/src/index.rs

```diff
@@ -47,21 +51,54 @@ pub struct Update {
 }
 
 impl Controller {
-    pub fn new(client: k8s::Client, updates: UnboundedReceiver<Update>) -> Self {
-        Self { client, updates }
+    pub fn new(
+        claims: Receiver<Arc<Claim>>,
+        client: k8s::Client,
+        name: String,
+        updates: UnboundedReceiver<Update>,
+    ) -> Self {
+        Self {
+            claims,
+            client,
+            name,
+            updates,
+            leader: false,
+        }
     }
 
-    pub async fn process_updates(mut self) {
-        let patch_params = k8s::PatchParams::apply("policy.linkerd.io");
+    pub async fn run(mut self) {
+        let patch_params = k8s::PatchParams::apply(POLICY_API_GROUP);
 
-        // todo: If an update fails we should figure out a requeueing strategy
-        while let Some(Update { id, patch }) = self.updates.recv().await {
-            let api =
-                k8s::Api::<k8s::policy::HttpRoute>::namespaced(self.client.clone(), &id.namespace);
+        // Select between the status-controller lease claim changing and
+        // receiving updates from the index. If the lease claim changes, then
+        // check if we are now the leader. If so, we should apply the patches
+        // received; otherwise, we should drain the updates queue but not
+        // apply any patches since another status controller is responsible
+        // for that.
+        loop {
+            tokio::select! {
+                biased;
+                _ = self.claims.changed() => {
+                    let claim = self.claims.borrow_and_update();
+                    if claim.is_current_for(&self.name) {
+                        self.leader = true
+                    } else {
+                        self.leader = false
+                    }
```

Comment:
small nit, not a big deal: i think this could be simplified to just
```suggestion
                    self.leader = claim.is_current_for(&self.name);
```